src/platform_impl/macos/window.rs

```diff
@@ -249,7 +249,7 @@ fn create_window(
                 let (x, y) = (increments.width, increments.height);
                 if x >= 1.0 && y >= 1.0 {
                     let size = NSSize::new(x as CGFloat, y as CGFloat);
-                    ns_window.setResizeIncrements_(size);
+                    ns_window.setContentResizeIncrements_(size);
```

Comment:
Any particular reason you chose [`contentResizeIncrements`](https://developer.apple.com/documentation/appkit/nswindow/1419649-contentresizeincrements?language=objc) over [`resizeIncrements`](https://developer.apple.com/documentation/appkit/nswindow/1419390-resizeincrements?language=objc)?

I'm guessing because it's easier for your use-case?

Perhaps the methods should be called `inner_resize_increments` and `set_inner_resize_increments`? See also https://github.com/rust-windowing/winit/issues/2308#issuecomment-1197248011.

Reply:
In the case of x11 specifically, it's really an "outer" size increment if you want to be more specific, since it's a hint to the window manager which willy typically handle resizing based on interaction with a separate parent frame that is created.

The distinction can even be important at times with x11 too because the outer size may change size out of sync with the inner size. To avoid artifacts window managers utilize a window "gravity" mechanism, and compositors try to hide the latency too. On a related note winit currently has some issues with how it handles this asynchronous behaviour on x11: https://github.com/rust-windowing/winit/issues/2306

Reply:
I'd maybe avoid being specific, with "outer" vs "inner" here since it may vary between window systems what the increments would be relative too.

Reply:
I think `resizeIncrements` is "outer" while `contentResizeIncrements` is "inner", so consistency is an argument for sticking with `resizeIncrements`

Reply:
On my desktop (awesomewm) if I set window borders to be super-thick resize increments behaved like 'inner' - and also inner increments make way more sense for the only application of resize increments I can think of - terminal emulators.

That's why I changed it to inner for the macos builder - also I vaguely recall that when I tried to implement this directly in alacritty a year ago the inner ones worked better on mac (that I then had at hands) or something.

Reply:
actually, yeah I suppose there is the sense that they are logically related to content - considering things like terminals or line-based views are the typical reason to want any resize increments

Reply:
tbh I think I was also misremembering how the wmhints work for x11 - it's one thing that the window manager is going to drive window resizing based on interaction with the outer frame, but the constraint it is going to be applying while handling a resize is indeed going to be to based on the desired increments for the inner window (as you say @necauqua - terminal emulators are the classic use case)

Reply:
Let's just go with `contentResizeIncrements`, maybe a short comment saying that we use that instead of `resizeIncrements` would be nice.

I think the name `resize_increments` is fine, it sounds like there's basically never a need for "outer" resize increments.